### PR TITLE
Update container images used in CI

### PR DESCRIPTION
**Pull request by bradfordb-vmware**
* Remove image for GP7 Ubuntu 18.04
* Remove image for GP7 RHEL 8
* Replace GP6 RHEL 8 with Rocky 8
* Replace rpmrebuild RHEL8 with Rocky 8

### CODE REVIEW

File: concourse/docker/diagram/README.md [hunk 1]
```diff
@@ -2,6 +2,225 @@
 
 ![container image building](./images.svg)
 
+```mermaid
+%%{init: {'theme':'neutral'}}%%
```
Comment (ashuka24) [inline concourse/docker/diagram/README.md:6]: if we already have the image as an svg and in `.dot` format why do we also need to include it again here?

File: concourse/docker/pxf-dev-base/cloudbuild.yaml [hunk 2]
```diff
@@ -72,29 +72,29 @@ steps:
 
 # Corresponds to the docker-gpdb-pxf-dev-rhel8 job in the docker pipeline
```
Comment (ashuka24) [inline concourse/docker/pxf-dev-base/cloudbuild.yaml:73]: this comments needs to be updated

File: concourse/docker/README.md [hunk 3]
```diff
@@ -15,45 +15,13 @@ changes to `pxf-build-base` and is also in charge of tagging the images as
...
-    <td> <a href="https://console.cloud.google.com/gcr/images/${GCR_PROJECT_ID}/GLOBAL/gpdb-pxf-dev/gpdb7-rocky8-test-pxf">gpdb7-rocky8-test-pxf</a> </td>
-  </tr>
-  <tr>
-    <td>MapR on CentOS7</td>
-    <td> <a href="https://console.cloud.google.com/gcr/images/${GCR_PROJECT_ID}/GLOBAL/gpdb-pxf-dev/gpdb6-centos7-test-pxf-mapr">gpdb6-centos7-test-pxf-mapr</a> </td>
-    <td> N/A </td>
-  </tr>
-</table>
-
-* Note: GCR_PROJECT_ID is the name of the Google Cloud Project ID
```
Comment (denalex) [inline concourse/docker/README.md:56]: seems we lost the links to the images, can we add the links back ?

File: concourse/docker/rpmrebuild/cloudbuild.yaml [hunk 4]
```diff
@@ -15,32 +15,19 @@ steps:
...
   id: rpmrebuild-rocky8-image
   args:
     - 'build'
-    - '--build-arg=BASE_IMAGE=${_BASE_IMAGE_REPOSITORY}/gpdb7-rocky8-test:latest'
+    - '--build-arg=BASE_IMAGE=rockylinux:8'
     - '--tag=gcr.io/$PROJECT_ID/rpmrebuild-rocky8:latest'
     - '-f'
-    - 'concourse/docker/rpmrebuild/rhel/Dockerfile'
+    - 'concourse/docker/rpmrebuild/rocky/Dockerfile'
     - 'concourse/docker/rpmrebuild/rhel'
```
Comment (denalex) [inline concourse/docker/rpmrebuild/cloudbuild.yaml:27]: should this be:
```suggestion
    - 'concourse/docker/rpmrebuild/rocky'
```

File: concourse/pipelines/templates/pr_pipeline-tpl.yml [hunk 5]
```diff
@@ -215,18 +211,18 @@ jobs:
...
   - task: Build PXF-GP[[gp_ver]] on RHEL8
-    image: gpdb[[gp_ver]]-pxf-dev-rhel8-image
+    image: gpdb[[gp_ver]]-pxf-dev-rocky8-image
     file: pxf_src/concourse/tasks/build.yml
     params:
       LICENSE: ((ud/pxf/common/rpm-license))
       VENDOR: ((ud/pxf/common/rpm-vendor))
   - task: Test Against Greenplum Latest RHEL8
-    image: gpdb[[gp_ver]]-pxf-dev-rhel8-image
+    image: gpdb[[gp_ver]]-pxf-dev-rocky8-image
```
Comment (denalex) [inline concourse/pipelines/templates/pr_pipeline-tpl.yml:225]: rocky8 was kinda a "secret" before, but now since Rocky8 is all over the place, should we rename everything that has rhel8 (job names, task names, resource names) to use rocky8 ?

File: concourse/docker/diagram/README.md (hunk 1 above)
Comment (denalex) [reply concourse/docker/diagram/README.md:6]: I find both of them somewhat hard to parse and update, but not having to run extra tools is beneficial.

File: concourse/docker/README.md (hunk 3 above)
Comment (bradfordb-vmware) [reply concourse/docker/README.md:56]: I removed the links since they didn't work as written because of using `${GCR_PROJECT_ID}` in the URL.

File: concourse/pipelines/templates/pr_pipeline-tpl.yml (hunk 5 above)
Comment (bradfordb-vmware) [reply concourse/pipelines/templates/pr_pipeline-tpl.yml:225]: We still use RHEL7 in the name of jobs and GPDB RPM resources even though our CI images are based on CentOS 7. I don't think we want to rename CI jobs because we will lose build history.

File: concourse/docker/diagram/README.md (hunk 1 above)
Comment (bradfordb-vmware) [reply concourse/docker/diagram/README.md]: I've removed the Grpahviz diagram and moved the Mermaid diagram to the docker README from a separate diagram README. I've also added a link to the Mermaid documentation for its syntax.

**Review by yimingli-vmware — APPROVED** [review]

**Review by ashuka24 — APPROVED** [review]

**Review by denalex — APPROVED** [review]
